website/docs/reference/resource-properties/constraints.md

```diff
@@ -68,9 +66,81 @@ models:
           - type: ...
 ```
 
+Today, in [dbt Cloud Versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless), you can use the new recommended syntax for defining foreign keys. This syntax leverages `ref`. For more information, refer to the section for the [forthcoming v1.9 release](/reference/resource-properties/constraints#whats-better-about-this-syntax).
+
+</File>
+
+</VersionBlock>
+
+<VersionBlock firstVersion="1.9">
+
+## Defining constraints
+
+Constraints may be defined for a single column, or at the model level for one or more columns. As a general rule, we recommend defining single-column constraints directly on those columns.
+
+If you are defining multiple `primary_key` constraints for a single model, those _must_ be defined at the model level. Defining multiple `primary_key` constraints at the column level is not supported. 
+
+The structure of a constraint is:
+- `type` (required): one of `not_null`, `unique`, `primary_key`, `foreign_key`, `check`, `custom`
+- `expression`: Free text input to qualify the constraint. Required for certain constraint types, and optional for others.
+- `name` (optional): Human-friendly name for this constraint. Supported by some data platforms.
+- `columns` (model-level only): List of column names to apply the constraint over
+
+Today, in [dbt Cloud Versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless), you can use the new recommended syntax for defining foreign keys. This syntax leverages `ref`.
+
+### What's better about this syntax?
+
+dbt will automatically resolve references to other models so that these references:
+
+- It captures dependencies.
+- Works across environments.
+
+Here are two examples demonstrating how the new syntax is utilized:
+
+<File name='models/schema.yml'>
```

Comment:
suggest being consistent with the code snippets. in 1.8 and lower, it's all one <File name> with the `model` and `column` indication, as opposed to two code snippets: